src/util-frontend.js

```diff
@@ -213,3 +213,96 @@ export function getToastErrorTimeout() {
 
     return errorTimeout;
 }
+
+class RelativeTimeFormatter {
+    /**
+     * Default locale and options for Relative Time Formatter
+     */
+    constructor() {
+        this.locale = currentLocale();
+        this.options = { numeric: "auto" };
+        this.rtf = new Intl.RelativeTimeFormat(this.locale, this.options);
+    }
+
+    /**
+     * Method to get the singleton instance
+     * @returns {object} Intl.RelativeTimeFormat instance
+     */
+    getInstance() {
+        return this.rtf;
+    }
+
+    /**
+     * Method to update the instance locale and options
+     * @param {string} locale Localization identifier (e.g., "en", "ar-sy") to update the instance with.
+     * @param {object} options Instance options to be passed.
+     * @returns {void} No return value.
+     */
+    updateLocale(locale, options = {}) {
+        this.locale = locale;
+        this.options = {
+            ...this.options,
+            ...options,
+        };
```

Comment:
I don't see why we would want to change the options on this. It is shared => would apply globally, right?

```suggestion
    updateLocale(locale) {
        this.locale = locale;
```

Reply:
Yes it shared and would be applied globally. Have removed it. Thanks!